src/Logging/EntityListener/AbstractEntityListener.php

```diff
@@ -111,11 +109,15 @@ public function createLogEntry(string $action, CaseEntity $case, LifecycleEventA
         $logEntry->setEntityID($object->getId());
         $logEntry->setAction($action);
 
+        /** @var User $user */
         $user = $this->security->getUser();
         if ($user) {
             $logEntry->setUser($user->getUsername());
         } else {
-            $logEntry->setUser('Fixtures');
+            $userRepository = $em->getRepository(User::class);
+            /** @var User $os2formsUser */
+            $os2formsUser = $userRepository->findOneBy(['name' => 'OS2Forms']);
+            $logEntry->setUser($os2formsUser->getName());
```

Comment:
I think we should authenticate as the OS2Forms user when updating stuff from OS2Forms to get rid of hacks and assumptions like this.

See https://gist.github.com/azhararmar/0a952cf03b1cfbd2a5b059089b764491?permalink_comment_id=3779292#gistcomment-3779292 for hints on how to programatically authenticate a user.

Reply:
👍 This also solves other 'hacky' stuff.